File: src/main/kotlin/fr/jhelp/kotlinLightSamples/ComparableTest.kt
package fr.jhelp.kotlinLightSamples

class ComparableTest : Comparable<ComparableTest>
{
    val string: String
    val number: Int

    constructor(string: String, number: Int)
    {
        this.string = string
        this.number = number
    }

    override fun compareTo(other: ComparableTest): Int
    {
        val comparison = this.string.compareTo(other.string)

        if (comparison != 0)
        {
            return comparison
        }

        return this.number - other.number
    }
}

File: src/main/swift/Tests/swiftTests/src/jhelp/kotlinLightSamples/ComparableTest.swift

public class ComparableTest : Comparable {

     static public func == (lhs: ComparableTest, rhs: ComparableTest) -> Bool
     {
          return lhs.compareTo(rhs) == 0
     }


     static public func < (lhs: ComparableTest, rhs: ComparableTest) -> Bool
     {
          return lhs.compareTo(rhs) < 0
     }


    let string: String
    let number: Int

    public init(_ string: String, _ number: Int){
        self.string = string
        self.number = number
    }

    public func compareTo(_ other: ComparableTest) -> Int
    {
        let comparison = self.string.compareTo(other.string)

        if comparison != 0
        {
            return comparison
        }

        return self.number - other.number
    }
}
